R/cdfPlot.R

```diff
@@ -264,54 +357,63 @@ cdfPlot <- function(contrastDF,
-            scale_size_manual(name = "Group", guide = "none", labels = ssc$group,
-                              values = ssc$symbolSize) +
-            scale_color_manual(name = "Group", guide = "none", labels = ssc$group,
-                               values = ssc$symbolColor) +
-            scale_fill_manual(name = "Group", guide = "none", labels = ssc$group,
-                              values = ssc$symbolFill) +
-            geom_rect(xmin = 0, xmax = subsetRows,
-                      ymin = 0, ymax = max(contrastDFsubset[[y]]), color = "lightblue",
+            scale_shape_manual(values = symbolShape) +
+            scale_size_manual( values = symbolSize) +
+            scale_color_manual(values = symbolColor, aesthetics = c("colour", "fill")) +
+            geom_rect(xmin = 0, xmax = nrow(contrastDF),
+                      ymin = 0, ymax = max(contrastDF[[y]]), color = "lightblue",
                       fill = "lightblue", alpha = 0.2) +
-            geom_point(alpha = alpha)
+            geom_point(alpha = transparency)
+
+        #remove the legends for the inset plot
+        cdfInset <- cdfInset + theme(legend.position = "none")
 
         # Add Labels and title
         cdfInset <- cdfInset +
             xlab(xlab) +
             ylab(ylab) +
             ggtitle(insetTitle)
 
-        # A viewport taking up a fraction of the plot area (upper left)
-        vp <- grid::viewport(width  = viewportWidth,
-                             height = viewportWidth,
-                             x      = viewportX,
-                             y      = viewportY,
-                             just   = c("left", "top"))
+        plot_limits <- get_plot_limits(cdfMain, viewportX, viewportY, viewportWidth)
+        vp_plot <- cdfMain +
+            annotation_custom(grob =  ggplotGrob(cdfInset),
+                              ymin = plot_limits[["ymin"]],
+                              ymax = plot_limits[["ymax"]],
+                              xmin = plot_limits[["xmin"]],
+                              xmax = plot_limits[["xmax"]])
 
-        if (printPlot) {
-            print(cdfMain)
-            print(cdfInset, vp = vp)
-        }
-
-        cdfPlot <- list(main = cdfMain, inset = cdfInset, viewport = vp)
+        cdfPlot <- list(main = cdfMain, inset = cdfInset, combined = vp_plot)
     }
+
     return(cdfPlot)
 }
+
+get_plot_limits <- function(main_plot, viewportX, viewportY, viewportWidth) {
+    main_plot_build <- ggplot_build(main_plot)
+    xrange <- main_plot_build$layout$panel_params[[1]]$x.range
+    yrange <- main_plot_build$layout$panel_params[[1]]$y.range
+
+    x_range_val <- xrange[[2]] - xrange[[1]]
+    xmin <- xrange[[1]] + (0.02 * x_range_val)
+    xmax <- xmin + (viewportWidth * x_range_val)
+
+    y_range_val <- yrange[[2]] - yrange[[1]]
+    ymin <- yrange[[2]] - (0.02 * y_range_val)
+    ymax <- ymin - (viewportWidth * y_range_val)
+    return(list("xmin" = xmin,
```

Comment:
no need for calling return function here, constructing the list will do the job and be returned as the last function statement

Reply:
done